csrc/includes/cpu_adagrad.h

```diff
@@ -130,6 +138,9 @@ void Adagrad_Optimizer::Step_AVX(size_t* rounded_size,
                                  ds_half_precision_t* dev_params,
                                  bool half_precision)
 {
+#if !defined(__AVX512__)
+    if (std::is_same_v<T, c10::BFloat16>) { return; }
```

Comment:
Why is it safe to return early and skip computation?

Reply:
Before returning, Step_AVX sets rounded_size=new_rounded_size so the remaining part of the buffer will be handled in non AVX fashion. returning early here would make the whole buffer fallback to non AVX path.